runtime/startup_aux.c

```diff
@@ -36,10 +36,36 @@ extern void caml_win32_unregister_overflow_detection (void);
 
 /* Initialize the atom table */
 
-CAMLexport header_t caml_atom_table[256];
+/* We add padding in the atom table to make sure it does not share a
+   page with a non-value (such as some bytecode), which would break
+   code which depend on the correctness of the page table.
+
+   Moreover, we add 1 padding at the end of the atom table since the
+   atom pointer actually points to the block *following* the entry in
+   the table.
+*/
```

Comment:
This comment is redundant with the comment below (starting with `PR#9128`).  Usually I'm the one who calls for more comments, but here I think you should remove the comment before the function and leave only the comment inside the function.  The alignment and padding business is an implementation issue (-> commented inside the function), not an interface issue (-> commented outside).